src/main/java/uk/gov/companieshouse/api/strikeoffobjections/service/impl/EmailService.java

```diff
@@ -0,0 +1,86 @@
+package uk.gov.companieshouse.api.strikeoffobjections.service.impl;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+import uk.gov.companieshouse.api.model.company.CompanyProfileApi;
+import uk.gov.companieshouse.api.strikeoffobjections.common.ApiLogger;
+import uk.gov.companieshouse.api.strikeoffobjections.email.KafkaEmailClient;
+import uk.gov.companieshouse.api.strikeoffobjections.model.email.EmailContent;
+import uk.gov.companieshouse.api.strikeoffobjections.service.ICompanyProfileService;
+import uk.gov.companieshouse.api.strikeoffobjections.service.IEmailService;
+import uk.gov.companieshouse.service.ServiceException;
+
+import java.time.LocalDateTime;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import java.util.function.Supplier;
+
+@Service
+public class EmailService implements IEmailService {
+
+    @Value("${EMAIL_SUBMITTED_EXTERNAL_TEMPLATE_APP_ID}")
+    private String originatingAppId;
```

Comment:
Perhaps give this variable a more specific name, in line with the one below.

Reply:
this is something i've been investigating this morning. We don't need to have more than one appId.
So I was going to suggest refactoring the configs to just be EMAIL_SENDER_APP_ID = "strike-off-objections"

As the email sender just checks that the appId is allowed to send the message type it's trying to send, so we can just allow "strike-off-objections" to send all the "types" for this service

Reply:
👍  yep, makes sense. I'll change that in my config PR no (not yet merged).

Reply:
Be warned this change would require a pr on https://github.com/companieshouse/chs-notification-api/blob/develop/assets/template_registry.yaml#L67
To accomodate. But I will update this pr with the more generic APP_ID env variable once you have updated your config pr

Reply:
Maybe @SVarallo can do that under his PR - https://github.com/companieshouse/chs-notification-api/pull/168 - which also isn't yet merged.

Reply:
Config PR updated - https://github.com/companieshouse/chs-configs/pull/1131. I think you can now change the key name in your code.